### PR TITLE
Explode possible renames for .window.js / .any.js test files

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Further work for #663 

Adds a utility for exploding possible renames for `.any.js`, which will improve rename handling for diffs.

### CODE REVIEW

**Comment by lukebjerring** [comment]
A merge of this and #933 shows the rename handling working:
https://temp-dot-wptdashboard-staging.appspot.com/results/workers?diff&run_id=5969501392732160&run_id=5487233003945984

File: shared/manifest.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import "strings"
+
+const (
+	windowJSSuffix                 = ".window.js"
+	windowHTMLSuffix               = ".window.html"
+	anyJSSuffix                    = ".any.js"
+	anyJSWindowHTMLSuffix          = ".any.html"
+	anyJSDedicatedWorkerHTMLSuffix = ".any.worker.html"
+	anyJSServiceWorkerHTMLSuffix   = ".any.serviceworker.html"
+	anyJSSharedWorkerHTMLSuffix    = ".any.sharedworker.html"
```
Comment (Hexcles) [inline shared/manifest.go]: There are in fact two types of different constants here: `.window.js` and `.any.js` are the keys, while the rest `*.html` are the expanded values, so perhaps we should define a map instead.

Besides, you are missing `.worker.js`.

Comment (lukebjerring) [reply shared/manifest.go]: Changed to a map + added worker, PTAL

File: shared/manifest.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+			result[eBefore[i]] = eAfter[i]
+		}
+	}
+	return result
+}
+
+// ExplodePossibleFilenames explodes the given single filename into the test names that
+// could be created for it at runtime.
+func ExplodePossibleFilenames(filePath string) []string {
+	// https://web-platform-tests.org/writing-tests/testharness.html#multi-global-tests
```
Comment (Hexcles) [inline shared/manifest.go]:
```suggestion
```

File: shared/manifest.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
+// Copyright 2018 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package shared
+
+import "strings"
+
+// explosions returns a map of the exploded test by filename suffix.
```
Comment (Hexcles) [inline shared/manifest.go]:
```suggestion
// explosions returns a map of the exploded test by filename suffix.
// https://web-platform-tests.org/writing-tests/testharness.html#multi-global-tests
```

File: shared/manifest.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+// ExplodePossibleRenames returns a map of equivalent renames for the given file rename.
+func ExplodePossibleRenames(before, after string) map[string]string {
+	result := map[string]string{
+		before: after,
+	}
+	eBefore := ExplodePossibleFilenames(before)
+	eAfter := ExplodePossibleFilenames(after)
+	if len(eBefore) == len(eAfter) {
```
Comment (Hexcles) [inline shared/manifest.go]: Optional: we could use more heuristics here if the lengths are different.

e.g. `.any.html` can be compared against `.window.html`, and `.any.worker.html` can be compared to `.worker.html`.

Comment (lukebjerring) [reply shared/manifest.go]: Might follow up with other cases.

**Pull request merged**